core/src/test/java/org/elasticsearch/indices/recovery/RecoveryTests.java

```diff
@@ -241,4 +246,38 @@ public void testPeerRecoveryPersistGlobalCheckpoint() throws Exception {
             assertThat(replica.getTranslog().getLastSyncedGlobalCheckpoint(), equalTo(numDocs - 1));
         }
     }
+
+    public void testSequenceBasedRecoveryKeepsTranslog() throws Exception {
```

Comment:
can you double check we have a test that make sure that all forms of recovery removes unneeded ops above the global checkpoint? if we don't think we can now add it?

Reply:
Yes. We have `RecoveryDuringReplicationTests#testRecoveryAfterPrimaryPromotion`. Previously this test was expected to execute a file-based sync, but now it will execute seq-based recovery.